File: BE/MIS_Project_Java/src/main/java/com/example/demo/SanPham.java
package com.example.demo;

import java.sql.Connection;
import java.sql.DatabaseMetaData;
import java.sql.DriverManager;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.sql.Statement;
import java.util.ArrayList;
import java.util.List;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

@JsonIgnoreProperties(value = { "all" })
public class SanPham {
	public String maSP;
	public String tenSP;
	public String anhSP;
	public String moTaSP;
	public String ngayDang;
	public String giaSP;
	public String soLuongTon;
	public String soSPDaBan;
	public String avgRating;
	public String soRating;
	public String loaiSP;
	public String account_CH;
	public String tenLoaiSP;
	public String maNhomSP;
	public String tenNhomSP;
	public String tenCH;
	
	public SanPham()
	{
		maSP = "";
		tenSP = "";
		anhSP = "";
		moTaSP = "";
		ngayDang = "";
		giaSP = "";
		soLuongTon = "";
		soSPDaBan = "";
		avgRating = "";
		soRating = "";
		loaiSP = "";
		account_CH = "";
		tenLoaiSP = "";
		maNhomSP = "";
		tenNhomSP = "";
		tenCH = "";
	}
	
	public SanPham(String MaSP, String TenSP, String AnhSP, String MoTaSP, String NgayDang, String GiaSP, String SoLuongTon, String SoSPDaBan, String AvgRating, String SoRating, String LoaiSP, String Account_CH, String TenLoaiSP, String MaNhomSP, String TenNhomSP, String TenCH)
	{
		this.maSP = MaSP;
		this.tenSP = TenSP;
		this.anhSP = AnhSP;
		this.moTaSP = MoTaSP;
		this.ngayDang = NgayDang;
		this.giaSP = GiaSP;
		this.soLuongTon = SoLuongTon;
		this.soSPDaBan = SoSPDaBan;
		this.avgRating = AvgRating;
		this.soRating = SoRating;
		this.loaiSP = LoaiSP;
		this.account_CH = Account_CH;
		this.tenLoaiSP = TenLoaiSP;
		this.maNhomSP = MaNhomSP;
		this.tenNhomSP = TenNhomSP;
		this.tenCH = TenCH;
	}
	
	public List<SanPham> getAll()
	{
		List<SanPham> products = new ArrayList<SanPham>();
		try 
		 {
			 String dbURL = "jdbc:sqlserver://localhost;databaseName=MIS;user=sa;password=t";
		     Connection conn = DriverManager.getConnection(dbURL);
		     if (conn != null) 
		     {
		         Statement stmt = conn.createStatement();
	             ResultSet rs = stmt.executeQuery("SELECT * FROM SanPham SP, LoaiSanPham LSP, NhomSanPham NSP, CuaHang CH WHERE SP.loaiSP = LSP.maLoaiSP and LSP.maNhomSP = NSP.maNhomSP and CH.AccountID = SP.Account_CH");
	             while (rs.next()) 
	             {
	            	 SanPham product = new SanPham(rs.getString(1), rs.getString(2), rs.getString(3), rs.getString(4), rs.getString(5),rs.getString(6), rs.getString(7), rs.getString(8), rs.getString(9), rs.getString(10),rs.getString(11), rs.getString(12), rs.getString(14), rs.getString(15), rs.getString(17), rs.getString(19));
	            	 products.add(product);
	             }
	             conn.close();
	             return products;
		     }
		 } 
		 catch (SQLException ex) 
		 {
			 System.err.println("Cannot connect database, " + ex);
		 }
		 	 return null;
	}
	
	public List<SanPham> getMany(String storeID)
	{
		List<SanPham> products = new ArrayList<SanPham>();
		try 
		 {
			 String dbURL = "jdbc:sqlserver://localhost;databaseName=MIS;user=sa;password=t";
		     Connection conn = DriverManager.getConnection(dbURL);
		     if (conn != null) 
		     {
		    	 DatabaseMetaData dm = (DatabaseMetaData) conn.getMetaData();
		         Statement stmt = conn.createStatement();
	             ResultSet rs = stmt.executeQuery("SELECT * FROM SanPham SP, LoaiSanPham LSP, NhomSanPham NSP, CuaHang CH WHERE SP.loaiSP = LSP.maLoaiSP and LSP.maNhomSP = NSP.maNhomSP and CH.AccountID = SP.Account_CH and CH.accountID = '" + storeID + "'");
	             while (rs.next()) 
	             {
	            	 SanPham product = new SanPham(rs.getString(1), rs.getString(2), rs.getString(3), rs.getString(4), rs.getString(5),rs.getString(6), rs.getString(7), rs.getString(8), rs.getString(9), rs.getString(10),rs.getString(11), rs.getString(12), rs.getString(14), rs.getString(15), rs.getString(17), rs.getString(19));
	            	 products.add(product);
	             }
	             conn.close();
	             return products;
		     }
		 } 
		 catch (SQLException ex) 
		 {
			 System.err.println("Cannot connect database, " + ex);
		 }
		 	 return null;
	}
}

File: BE/MIS_Project_C/MIS_Project/Models/SanPham.cs
using System;
using System.Collections.Generic;
using System.Data.SqlClient;
using System.Linq;
using System.Threading.Tasks;

namespace MIS_Project.Models
{
    public class SanPham
    {
        public string MaSP { get; set; }
        public string TenSP { get; set; }
        public string AnhSP { get; set; }
        public string MoTaSP { get; set; }
        public string NgayDang { get; set; }
        public string GiaSP { get; set; }
        public string SoLuongTon { get; set; }
        public string SoSPDaBan { get; set; }
        public string AvgRating { get; set; }
        public string SoRating { get; set; }
        public string LoaiSP { get; set; }
        public string Account_CH { get; set; }
        public string TenLoaiSP { get; set; }
        public string MaNhomSP { get; set; }
        public string TenNhomSP { get; set; }
        public string TenCH { get; set; }


        public List<SanPham> getAll()
        {
            SqlCommand com = new SqlCommand();
            SqlDataReader dr;
            SqlConnection con = new SqlConnection();
            con.ConnectionString = MIS_Project.Properties.Resources.ConnectionString;
            List<SanPham> products = new List<SanPham>();
            try
            {
                con.Open();
                com.Connection = con;
                com.CommandText = "SELECT * FROM SanPham SP, LoaiSanPham LSP, NhomSanPham NSP, CuaHang CH WHERE SP.loaiSP = LSP.maLoaiSP and LSP.maNhomSP = NSP.maNhomSP and CH.AccountID = SP.Account_CH";
                dr = com.ExecuteReader();
                while (dr.Read())
                {
                    products.Add(new SanPham()
                    {
                        MaSP = dr["MaSP"].ToString(),
                        TenSP = dr["TenSP"].ToString(),
                        AnhSP = dr["AnhSP"].ToString(),
                        MoTaSP = dr["MoTaSP"].ToString(),
                        NgayDang = dr["NgayDang"].ToString(),
                        GiaSP = dr["GiaSP"].ToString(),
                        SoLuongTon = dr["SoLuongTon"].ToString(),
                        SoSPDaBan = dr["SoSPDaBan"].ToString(),
                        AvgRating = dr["AvgRating"].ToString(),
                        SoRating = dr["SoRating"].ToString(),
                        LoaiSP = dr["LoaiSP"].ToString(),
                        TenLoaiSP = dr["TenLoaiSP"].ToString(),
                        MaNhomSP = dr["MaNhomSP"].ToString(),
                        TenNhomSP = dr["TenNhomSP"].ToString(),
                        Account_CH = dr["Account_CH"].ToString(),
                        TenCH = dr["TenCH"].ToString()
                    });
                }
                con.Close();
            }
            catch (Exception exc)
            {
                throw exc;
            }
            return products;
        }


        public List<SanPham> getOne(string productID)
        {
            SqlCommand com = new SqlCommand();
            SqlDataReader dr;
            SqlConnection con = new SqlConnection();
            con.ConnectionString = MIS_Project.Properties.Resources.ConnectionString;
            List<SanPham> products = new List<SanPham>();
            try
            {
                con.Open();
                com.Connection = con;
                com.CommandText = "SELECT * FROM SanPham SP, LoaiSanPham LSP, NhomSanPham NSP, CuaHang CH WHERE CH.AccountID = SP.Account_CH and SP.loaiSP = LSP.maLoaiSP and LSP.maNhomSP = NSP.maNhomSP and SP.maSP = '" + productID + "'";
                dr = com.ExecuteReader();
                while (dr.Read())
                {
                    products.Add(new SanPham()
                    {
                        MaSP = dr["MaSP"].ToString(),
                        TenSP = dr["TenSP"].ToString(),
                        AnhSP = dr["AnhSP"].ToString(),
                        MoTaSP = dr["MoTaSP"].ToString(),
                        NgayDang = dr["NgayDang"].ToString(),
                        GiaSP = dr["GiaSP"].ToString(),
                        SoLuongTon = dr["SoLuongTon"].ToString(),
                        SoSPDaBan = dr["SoSPDaBan"].ToString(),
                        AvgRating = dr["AvgRating"].ToString(),
                        SoRating = dr["SoRating"].ToString(),
                        LoaiSP = dr["LoaiSP"].ToString(),
                        TenLoaiSP = dr["TenLoaiSP"].ToString(),
                        MaNhomSP = dr["MaNhomSP"].ToString(),
                        TenNhomSP = dr["TenNhomSP"].ToString(),
                        Account_CH = dr["Account_CH"].ToString(),
                        TenCH = dr["TenCH"].ToString()
                    });
                }
                con.Close();
            }
            catch (Exception exc)
            {
                throw exc;
            }
            return products;
        }

        public List<SanPham> getMany(string subID)
        {
            SqlCommand com = new SqlCommand();
            SqlDataReader dr;
            SqlConnection con = new SqlConnection();
            con.ConnectionString = MIS_Project.Properties.Resources.ConnectionString;
            List<SanPham> products = new List<SanPham>();
            try
            {
                con.Open();
                com.Connection = con;
                com.CommandText = "SELECT * FROM SanPham SP, LoaiSanPham LSP, NhomSanPham NSP, CuaHang CH WHERE CH.AccountID = SP.Account_CH and SP.loaiSP = LSP.maLoaiSP and LSP.maNhomSP = NSP.maNhomSP and SP.loaiSP = '" + subID + "'";
                dr = com.ExecuteReader();
                while (dr.Read())
                {
                    products.Add(new SanPham()
                    {
                        MaSP = dr["MaSP"].ToString(),
                        TenSP = dr["TenSP"].ToString(),
                        AnhSP = dr["AnhSP"].ToString(),
                        MoTaSP = dr["MoTaSP"].ToString(),
                        NgayDang = dr["NgayDang"].ToString(),
                        GiaSP = dr["GiaSP"].ToString(),
                        SoLuongTon = dr["SoLuongTon"].ToString(),
                        SoSPDaBan = dr["SoSPDaBan"].ToString(),
                        AvgRating = dr["AvgRating"].ToString(),
                        SoRating = dr["SoRating"].ToString(),
                        LoaiSP = dr["LoaiSP"].ToString(),
                        TenLoaiSP = dr["TenLoaiSP"].ToString(),
                        MaNhomSP = dr["MaNhomSP"].ToString(),
                        TenNhomSP = dr["TenNhomSP"].ToString(),
                        Account_CH = dr["Account_CH"].ToString(),
                        TenCH = dr["TenCH"].ToString()
                    });
                }
                con.Close();
            }
            catch (Exception exc)
            {
                throw exc;
            }
            return products;
        }

        public SanPham Rating(SanPham Product)
        {
            SqlCommand com = new SqlCommand();
            SqlDataReader dr;
            SqlConnection con = new SqlConnection();
            con.ConnectionString = MIS_Project.Properties.Resources.ConnectionString;
            try
            {
                con.Open();
                com.Connection = con;
                com.CommandText = "UPDATE SanPham SET avgRating = ROUND(((avgRating * soRating) + " + Product.AvgRating + ")/(soRating + 1),2), soRating = soRating + 1 WHERE maSP = '" + Product.MaSP + "'";
                com.ExecuteNonQuery();
                con.Close();
            }
            catch (Exception exc)
            {
                throw exc;
            }
            return getOne(Product.MaSP)[0];
        }

    }
}
